test/link/glibc_compat/glibc_runtime_check.zig

```diff
@@ -0,0 +1,95 @@
+// A zig test case that exercises some glibc symbols that have uncovered
+// problems in the past.  This test must be compiled against a glibc.
+//
+// This does cannot test if the symbols are actually dynamically linked
+// (i.e., that they're not erroneously statically linked).
+
+const std = @import("std");
+const builtin = @import("builtin");
+const assert = std.debug.assert;
+
+const c_malloc = @cImport(
+    @cInclude("malloc.h"), // for reallocarray
+);
+
+const c_stdlib = @cImport(
+    @cInclude("stdlib.h"), // for atexit
+);
+
+// Grab the version of glibc targetted by Zig
+const major = builtin.target.os.version_range.linux.glibc.major;
+const minor = builtin.target.os.version_range.linux.glibc.minor;
+
+// PR #17034 - fstat moved between libc_nonshared and libc
+fn checkStat() !void {
+    const cwdFd = std.fs.cwd().fd;
+
+    var stat = std.mem.zeroes(std.c.Stat);
+    var result = std.c.fstatat(cwdFd, "a_file_that_definitely_does_not_exist", &stat, 0);
+    assert(result == -1);
+    assert(std.c.getErrno(result) == .NOENT);
+
+    result = std.c.stat("a_file_that_definitely_does_not_exist", &stat);
+    assert(result == -1);
+    assert(std.c.getErrno(result) == .NOENT);
+}
+
+// PR #17607 - reallocarray not visible in headers
+inline fn checkReallocarray() !void {
```

Comment:
don't mark functions as `inline`

Reply:
Ah, I thought `inline` was necessary to make the version check happen at comptime.  I've dropped the `inline` and added an explicit `comptime` on the version comparison expression.